app/utils.server.ts

```diff
@@ -36,3 +36,24 @@ export const doesFileExist = async (filePath: string): Promise<boolean> => {
     throw err;
   }
 };
+
+export const runOnTimeout = async <T>(
+  opts: { waitFor: Promise<T>; timeoutMs: number },
+  runOnTimeoutFn: () => void,
```

Comment:
Perhaps this should be extended to async functions?

Reply:
Sure, but I suggest changing it in the future when it's needed.